llvm/lib/Target/RISCV/RISCVTargetMachine.cpp

```diff
@@ -100,6 +100,10 @@ static cl::opt<bool> EnableMISchedLoadClustering(
     cl::desc("Enable load clustering in the machine scheduler"),
     cl::init(false));
 
+static cl::opt<bool> EnableVSETVLIAfterRVVRegAlloc(
+    "riscv-vsetvli-after-rvv-regalloc", cl::Hidden,
+    cl::desc("vsetvl insertion after rvv regalloc"), cl::init(true));
```

Comment:
Should we split this into 2 patches. One where this defaults to false with some directed tests that set it to true on the command line. Then a second patch that changes the default?

Reply:
I revert the testcase change and default disable `riscv-vsetvli-after-rvv-regalloc` in this patch. And add more testcase from  other lit that generate worse codegen into `splitRA-vsetvl.ll`.